### DISCUSSION TITLE
Problem customising routes

### DISCUSSION THREAD

**Discussion by alvaroo278** [c0c0n3/kitt4sme.live#191] [General]
Hello, I am trying to customise the access to the services in the web browser through routes, so as not to have to be accessed through the port, the same as for accessing argocd, it is through "http://hostname/argocd" and redirects you to the login for this I have modified the file "mesh-infra/routing/ingress".

![image](https://user-images.githubusercontent.com/74782660/219409597-ed1c4472-ead1-4c79-a0c1-5a96fbd50395.png)

The problem is that when I try to access through the indicated route, I get an authorisation error "Http Error 403".

I have not changed anything in terms of Keycloak, Grafana or Crate configuration.

**Comment by c0c0n3** [comment]
Hi @alvaroo278 :-)

> get an authorisation error "Http Error 403".

That's because of platform security rules. But since security isn't a requirement anymore for open calls, the easiest way to get around this is to disable security for those routes, as in the example below for Argo CD

- https://github.com/c0c0n3/kitt4sme.live/blob/main/deployment/mesh-infra/security/ingress-policy.yaml#L26-L27

For the record, Argo CD is excluded from the OPA-validated resources b/c it already comes with its own security & SSO.

**Reply by alvaroo278** [reply]
Ohh thanks!!

Then If security isn't a requirement for open calls, Can I remove opa, no?

**Reply by alvaroo278** [reply]
Hi @c0c0n3  :)

So If I wanna that when write "http://hostname/grafana" or  "http://hostname/crate", give me the same resource if I write "http://hostname:3000" or  "http://hostname:4200". To access the service from the browser without entering the port.

Thanks for help :)

**Comment by c0c0n3** [comment]
> Can I remove opa
 
Yep, but the easiest thing to do is probably just exclude the ingress policy from the Kustomize build

- https://github.com/c0c0n3/kitt4sme.live/blob/main/deployment/mesh-infra/security/kustomization.yaml#L5

and the delete the ingress policy (AuthorizationPolicy CRD) from your cluster.

> write "http://hostname/grafana" or "http://hostname/crate", give me the same resource if I write "http://hostname:3000/" or "http://hostname:4200/"

besides getting rid of the ingress policy and configuring path-based routing, you also might have to tweak crate and grafana to work with path-based routing, see e.g.

- https://grafana.com/tutorials/run-grafana-behind-a-proxy/
